3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -386,6 +407,45 @@ object EnrichmentManager {
       case _ => unitSuccess
     }
 
+    // This enrichment cannot fail
+    (registry.getEventFingerprintEnrichment match {
+      case Some(efe) => event.event_fingerprint = efe.getEventFingerprint(sourceMap)
+      case _ => ()
+    })
+
+    // Validate contexts and unstructured events
+    val shred = Shredder.shred(event) match {
+      case Failure(msgs) => msgs.map(_.toString).fail
+      case Success(_) => unitSuccess.toValidationNel
+    }
+
+    // Extract the event vendor/name/format/version
+    val extractSchema = SchemaEnrichment.extractSchema(event).map(schemaKey => {
+      event.event_vendor = schemaKey.vendor
+      event.event_name = schemaKey.name
+      event.event_format = schemaKey.format
+      event.event_version = schemaKey.version
+      unitSuccess
+    })
+
+    val derivedTstamp = event.true_tstamp match {
```

Comment:
The true_tstamp overriding the other timestamps to determine the final derived timestamp is part of the derived timestamp calculation algorithm - therefore that logic switch should be inside the EE.getDerivedTimestamp function, not exposed out here in EnrichmentManager. The added advantage of this is that this behaviour is then unit-testable in the DerivedTimestampSpec.